event-handlers/Slate/CBL/Demonstrations/DemonstrationSkill/afterRecordSave/50_complete-competency.php

```diff
@@ -20,6 +20,11 @@
         'StudentID' => $Student->ID,
         'CompetencyID' => $Competency->ID,
         'Level' => $completion['currentLevel'] + 1,
-        'EnteredVia' => 'graduation'
+        'EnteredVia' => 'graduation',
+        'BaselineRating' => $completion['demonstrationsAverage']
     ], true);
+} elseif (empty($completion['baselineRating'])) {
+    if ($StudentCompetency = $Competency->getStudentCompetency($Student)) {
+        $StudentCompetency->calculateStartingRating();
```

Comment:
`calculateStartingRating` is supposed to return a value, not mutate state itself. The writing and saving if it comes back non-null should happen here. The verb "calculate" doesn't sound like a command to mutate state